data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconBlockHandler.java

```diff
@@ -15,60 +15,79 @@
 
 import static tech.pegasys.artemis.datastructures.util.BeaconStateUtil.compute_start_slot_at_epoch;
 
+import com.google.common.eventbus.EventBus;
 import com.google.common.primitives.UnsignedLong;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import org.apache.commons.lang3.tuple.Pair;
 import org.apache.tuweni.bytes.Bytes32;
 import tech.pegasys.artemis.beaconrestapi.handlerinterfaces.BeaconRestApiHandler;
 import tech.pegasys.artemis.datastructures.blocks.BeaconBlock;
+import tech.pegasys.artemis.datastructures.blocks.SignedBeaconBlock;
 import tech.pegasys.artemis.storage.ChainStorageClient;
+import tech.pegasys.artemis.storage.HistoricalChainData;
 
 public class BeaconBlockHandler implements BeaconRestApiHandler {
 
   private final ChainStorageClient client;
+  private final HistoricalChainData historicalChainData;
 
-  public BeaconBlockHandler(ChainStorageClient client) {
+  public BeaconBlockHandler(ChainStorageClient client, EventBus eventBus) {
     this.client = client;
+    this.historicalChainData = new HistoricalChainData(eventBus);
```

Comment:
We should be passing in the `HistoricalChainData` client, not creating it in the constructor here.  Possibly should even be passing `HistoricalChainData` into `BeaconRestApi` rather than event bus.

Reply:
Done.